File: Data Structures & Algorithms/two-integer-sum/submission-12.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        my_map = {}

        for index in range(len(nums)):
            result = target - nums[index]

            if result in my_map:
                return [my_map.get(result), index]
            
            my_map[nums[index]] = index
        
        return [-1, -1]

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-12.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int [] leftArray = new int [nums.length];
        int [] rightArray = new int [nums.length];
        int [] result = new int [nums.length];

        int value = 1;

        for(int i = 0; i < nums.length; i++){
            leftArray[i] = value;
            value *= nums[i];
        }

        value = 1;

        for(int i = nums.length -1; i >= 0; i--){
            rightArray[i] = value;
            value *= nums[i];
        }

        for(int i = 0; i < nums.length; i++){
            result[i] = leftArray[i] * rightArray[i];
        }

        return result;
    }
}  
